### PR TITLE
Feature: Add wallet retired status

### PR DESCRIPTION
**Pull request by crm-skyach**
Display address from active wallets

Generate address from active wallets

deposit/withdraw from retired wallets

Add status for address

### CODE REVIEW

File: app/models/deposit.rb [hunk 1]
```diff
@@ -183,6 +183,7 @@ def as_json_for_event_api
       currency:                 currency_id,
       amount:                   amount.to_s('F'),
       state:                    aasm_state,
+      wallet_state:             PaymentAddress.find_by_address(address)&.wallet&.status,
```
Comment (chumaknadya) [inline app/models/deposit.rb]: PaymentAddress.find_by_address(address).wallet.status

You dont need & here

File: app/models/wallet.rb [hunk 2]
```diff
@@ -105,6 +106,10 @@ def kinds(options={})
     end
 
     def deposit_wallet(currency_id)
```
Comment (chumaknadya) [inline app/models/wallet.rb:112]: do you still need this function?
Is there any place where we use it?

**Review by chumaknadya — CHANGES_REQUESTED** [review]

File: app/models/deposit.rb (hunk 1 above)
Comment (crm-skyach) [reply app/models/deposit.rb]: Hi @chumaknadya i have removed &

File: app/models/wallet.rb (hunk 2 above)
Comment (crm-skyach) [reply app/models/wallet.rb:112]: Hi @chumaknadya this function is returning deposit wallets from active and retired wallets. We are using this function on several places.

**Review by chumaknadya — APPROVED** [review]

File: app/models/wallet.rb [hunk 3]
```diff
@@ -105,6 +106,10 @@ def kinds(options={})
...
+      Wallet.active_retired.deposit.joins(:currencies).find_by(currencies: { id: currency_id })
+    end
+
+    def active_wallet(currency_id)
```
Comment (mnaichuk) [inline app/models/wallet.rb]: better to name this method like `actve_deposit_wallet`

Comment (crm-skyach) [reply app/models/wallet.rb]: Hi @mnaichuk i have changed the name

File: app/services/wallet_service.rb [hunk 4]
```diff
@@ -29,7 +29,7 @@ def spread_deposit(deposit)
                        currency: deposit.currency.to_blockchain_api_settings)
 
     destination_wallets =
-      Wallet.active.withdraw.ordered
+      Wallet.active_retired.withdraw.ordered
```
Comment (chumaknadya) [inline app/services/wallet_service.rb]: here should be active

File: app/workers/amqp/withdraw_coin.rb [hunk 5]
```diff
@@ -40,7 +40,7 @@ def process(payload)
                        rid: withdraw.rid,
                        message: 'Sending witdraw.'
 
-          wallet = Wallet.active.joins(:currencies)
+          wallet = Wallet.active_retired.joins(:currencies)
```
Comment (chumaknadya) [inline app/workers/amqp/withdraw_coin.rb]: here should be just active

File: app/workers/daemons/deposit.rb [hunk 6]
```diff
@@ -81,7 +81,7 @@ def collect_fee(deposit)
           Rails.logger.warn { "The deposit was spreaded in the next way: #{deposit.spread}"}
         end
 
-        fee_wallet = Wallet.active.fee.find_by(blockchain_key: deposit.currency.blockchain_key)
+        fee_wallet = Wallet.active_retired.fee.find_by(blockchain_key: deposit.currency.blockchain_key)
```
Comment (chumaknadya) [inline app/workers/daemons/deposit.rb]: should be 'active' here

File: app/workers/daemons/deposit.rb [hunk 7]
```diff
@@ -81,7 +81,7 @@ def collect_fee(deposit)
           Rails.logger.warn { "The deposit was spreaded in the next way: #{deposit.spread}"}
```
Comment (chumaknadya) [inline app/workers/daemons/deposit.rb]: in line 54
we should have more complex logic
In deposit you have payment address, and you need to get your wallet from the payment address here
like payment_address.wallet

**Review by chumaknadya — CHANGES_REQUESTED** [review]

File: app/services/wallet_service.rb (hunk 4 above)
Comment (crm-skyach) [reply app/services/wallet_service.rb]: done

File: app/workers/amqp/withdraw_coin.rb (hunk 5 above)
Comment (crm-skyach) [reply app/workers/amqp/withdraw_coin.rb]: changed this to active

File: app/workers/daemons/deposit.rb (hunk 6 above)
Comment (crm-skyach) [reply app/workers/daemons/deposit.rb]: change this to active

File: app/workers/daemons/deposit.rb (hunk 7 above)
Comment (crm-skyach) [reply app/workers/daemons/deposit.rb]: done the changes

File: app/models/deposit.rb [hunk 8]
```diff
@@ -176,13 +176,18 @@ def uid=(uid)
     self.member = Member.find_by_uid(uid)
   end
 
+  def wallet_state
+    currency.coin? ? PaymentAddress.find_by_address(address).wallet.status : ''
```
Comment (chumaknadya) [inline app/models/deposit.rb]: PaymentAddress.find_by_address(address).wallet&.status

File: app/models/payment_address.rb [hunk 9]
```diff
@@ -47,6 +47,10 @@ def to_cash_address
     CashAddr::Converter.to_cash_address(address)
   end
 
+  def status
+    address.present? ? wallet.status : 'pending'
```
Comment (chumaknadya) [inline app/models/payment_address.rb]: wallet&.status

File: app/workers/daemons/deposit.rb [hunk 10]
```diff
@@ -10,7 +10,7 @@ def process
         ::Deposit.processing.each do |deposit|
           Rails.logger.info { "Starting processing coin deposit with id: #{deposit.id}." }
 
-          wallet = Wallet.deposit_wallet(deposit.currency_id)
+          wallet = Wallet.active_deposit_wallet(deposit.currency_id)
```
Comment (chumaknadya) [inline app/workers/daemons/deposit.rb]: wallet = PaymentAddress.find_by(address: deposit.address).wallet

**Review by chumaknadya — CHANGES_REQUESTED** [review]

**Review by mnaichuk — APPROVED** [review]

**Pull request merged**